Add 'systematic_name', queries fallback on 'common_name'

Im-0xea opened the pull request:
Hyia!

I recently added the 'systematic_name' property on the wiki so that the API should be able to provide the preferred IUPAC name once appropriate tagging is done, which I would gladly be responsible for.

besides that I did some minor fixes and made queries fall back to common_names if no page exists, so that with slighly more processing broad searches can cover alot more ground.

Let me know what you think!

Lots of love,
Xea

Im-0xea commented:
![20240516_03h45m15s_grim](https://github.com/psychonautwiki/bifrost/assets/136126731/7386917d-b7a2-42b4-a00b-9f52130682e9)

19h commented:
This is a breaking change and very likely breaks our api consumers, which isn't exactly good. What's the problem with chemical class et al.?

Im-0xea commented:
@19h I think the formating of the diff made you missunderstand my changes

there is no problem with "chemical class", I have not touched "chemical class", I removed a duplicate function that I found, and added a new property for systematic IUPAC names, that should be a breaking change

otherwise I added a fallback to the query function to cover more ground, I hope that doesn't disrupt anything, if so I would just revert that commit

19h commented:
![CleanShot 2024-05-28 at 19 48 54@2x](https://github.com/psychonautwiki/bifrost/assets/280212/96aef666-f1c5-4341-85a9-aa2f977bad15)

This looks a bit more than just formatting right?

Im-0xea commented:
@19h please look at the full file, I overrode a duplicate function for formating "common_name", if you look just a bit above the snippet you just send me you will see that the "chemical_class" function is still there

alevkov commented:
just rewrite in rust
